cacheops/redis.py

```diff
@@ -25,115 +42,74 @@ def handle_connection_failure(call):
             warnings.warn("The cacheops cache timed out! Error: %s" % e, RuntimeWarning)
         except Exception as e:
             warnings.warn("".join(traceback.format_exception(*sys.exc_info())))
-else:
-    handle_connection_failure = identity
 
-client_class_name = getattr(settings, 'CACHEOPS_CLIENT_CLASS', None)
-client_class = import_string(client_class_name) if client_class_name else redis.StrictRedis
-read_clients = []
+    class SafeRedis(client_class):
+        get = handle_connection_failure(client_class.get)
 
+    client_class = SafeRedis
 
-def ip(url):
-    try:
-        return socket.gethostbyname(urlparse(url).hostname)
-    except socket.gaierror:
-        warnings.warn('Hostname in URL %s did not resolve' % url)
-        raise
 
+if settings.CACHEOPS_REDIS_REPLICAS:
 
-def set_read_clients():
-    primary_url = settings.REDIS_MASTER
-    primary_ip = ip(primary_url)
-    replica_weight = settings.REDIS_REPLICA_WEIGHT
+    class ReplicaProxyRedis(client_class):
 
-    redis_urls = settings.REDIS_REPLICAS
-    # the conf could be a list or string
-    # list would look like: ["redis://cache-001:6379/1", "redis://cache-002:6379/2"]
-    # string would be: "redis://cache-001:6379/1,redis://cache-002:6379/2"
-    if isinstance(redis_urls, six.string_types):
-        redis_urls = redis_urls.split(',')
-    else:
-        redis_urls = list(redis_urls)
+        read_clients = []
 
-    # Make Redis clients from all the URLs except the primary
-    new_read_clients = [redis.StrictRedis.from_url(u) for u in redis_urls if ip(u) != primary_ip]
+        @classmethod
+        def set_read_clients(cls):
+            primary_ip = ip(settings.CACHEOPS_REDIS['host'])
+            replicas = settings.CACHEOPS_REDIS_REPLICAS
+            replica_weight = settings.CACHEOPS_REPLICA_WEIGHT
```

Comment:
We were using the non-cacheops settings `REDIS_MASTER` and `REDIS_REPLICAS`. Now we use the ones set up for this library, which are always `dict`s not URL strings.